mongo/changelog.d/17564.added

```diff
@@ -0,0 +1,6 @@
+Emit updated mongodb_instance event when mongo deployment type is refreshed with updates. The mongo deployment type is considered to be updated when
+- replica set name is changed
+- member role is updated in a replica set, i.e. primary step down/secondary step up or new member joins the replica set
+- new shard joins a sharded cluster or new member joins a shard results in mongos shard map updated
+Add internal resource tag `dd.internal.resource:database_instance` to mongo metrics.
```

Comment:
just a nit, but this isn't very customer facing friendly. I'm torn on whether it should even be mentioned in the changelog, since it kind of reveals how the sausage is made (with tagging)

Reply:
make sense. will remove